tests/unit/Internal/Primitives/CallableTypeTest.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Facile\PhpCodec\Internal\Primitives;
+
+use Eris\TestTrait;
+use Facile\PhpCodec\Codecs;
+use Tests\Facile\PhpCodec\BaseTestCase;
+use Tests\Facile\PhpCodec\GeneratorUtils;
+
+class CallableTypeTest extends BaseTestCase
+{
+    use TestTrait;
+
+    public function testLaws(): void
+    {
+        $this
+            ->forAll(
+                GeneratorUtils::scalar(),
+                static function (): void {
+                },
+                static function (int $a) {
+                    return $a;
+                }
+            )
+            ->then(self::codecLaws(Codecs::callable()));
+    }
```

Comment:
There are [several ways to represent something callable in Php](https://www.php.net/manual/en/language.types.callable.php), but this test is taking into account only anonymous functions.
Could we add more tests (or improve the current generators) to cover more cases?